txmongo/collection.py

```diff
@@ -449,6 +466,8 @@ def after_reply(reply, protocol, this_func, fetched=0):
                     to_fetch = limit - fetched
                     if to_fetch <= 0:
                         to_fetch = None  # close cursor
+                    elif batchsize:
```

Comment:
We are trying hard to cover as much code branches as possible to avoid regressions. Could you please add some tests covering all these branches in `__real_find_with_cursor`?